product/views.py

```diff
@@ -1,59 +1,154 @@
 import json
+from django.utils import timezone
+import datetime
 
 from django.views     import View
 from django.http      import JsonResponse
+from django.db.models import Prefetch, Count
 
 from .models import (
     Product,
     ProductImage,
     MainCategory,
-    ProductCategory
+    ProductCategory,
+    ProductComment,
 )
 
-class ProductListView(View):
-    def get(self, request, type):
-            category_seq    = request.GET.get('category', None)
-            subcategory_seq = request.GET.get('subcategory', None)
-            sort            = request.GET.get('sort', None)
-
-            if type == "new":
-                products = Product.objects.filter(Q(category=category_seq) | Q (subcategory=subcategory_seq), id__range=(1,8)).order_by(sort).values()
-            elif type == "all":
-                products = Product.objects.filter(Q(category=category_seq) | Q (subcategory=subcategory_seq), id__range=(1,40)).order_by(sort).values()
-
-            filtered_products = [{
-                "id"            : item["id"],
-                "name"          : item["name"],
-                "price"         : int(item["price"]),
-                "star_rating"   : item["star_rating"],
-                "description"   : item["description"],
-                "category_id"   : item["category_id"],
-                "subcategory_id": item["subcategory_id"],
-                "discount_id"   : item["discount_id"],
-                "image_url"     : list(ProductImage.objects.filter(product_id = item["id"]).values_list('image_url', flat=True)),
-                "created_at"    : item["created_at"],
-            }for item in products]
-
-            return JsonResponse({"message" : "SUCCESS", "result" : filtered_products}, status = 200)
-            
+from .forms import ProductFullForm
+
+from user.models import OrderStatus, Address, User, Wishlist
+
+import boto3          
+import uuid   
+
+
+class ProductsView(View):
+    def get(self, request):
+        address_id = request.GET.get('address_id', None)
+
+        try:
+            if Product.objects.filter(address=address_id).exists():
+                products = Product.objects.filter(address=address_id).prefetch_related(Prefetch('productimage_set', queryset = ProductImage.objects.all(), to_attr='productimage_list'),
+                                                                                        Prefetch('wishlist_set', queryset = Wishlist.objects.all(), to_attr='wishlist_list'),
+                                                                                        Prefetch('productcomment_set', queryset = ProductComment.objects.all(), to_attr='comment_list'))   
```

Comment:
장고 공식 문서를 보고 M:1 관계에서 prefetch_related를 썼습니다. 또한, Product의 관점에서 ProductImage는 역참조의 관계이기 때문에 Prefetch_related를 썼는데 제가 이해를 잘못하고 있는 부분이 있나요??ㅠㅜ

select_related works by creating an SQL join and including the fields of the related object in the SELECT statement. For this reason, select_related gets the related objects in the same database query. However, to avoid the much larger result set that would result from joining across a ‘many’ relationship, **select_related is limited to single-valued relationships - foreign key and one-to-one.**

prefetch_related, on the other hand, does a separate lookup for each relationship, and does the ‘joining’ in Python. **This allows it to prefetch many-to-many and many-to-one objects,** which cannot be done using select_related, in addition to the foreign key and one-to-one relationships that are supported by select_related.

Reply:
승연님 제가 모델을 잘 못 생각했어요~
역참조 관계라면 prefetch_related가 맞습니다~